user_notification/models.py

```diff
@@ -24,22 +26,58 @@ class JobAlert(models.Model):
     def __str__(self):
         return str(self.user_account_id)
 
-    # Calculate the date of the next alert
     @classmethod
     def calc_check_date(cls, last_check_date, frequency_in_days):
+        # Calculate the date of the next alert
         if frequency_in_days < 1:
-            myError = ValueError(
+            raise ValueError(
                 'frequency_in_days should be Greater than or equal to 1')
-            raise myError
         else:
-            if isinstance(last_check_date, datetime):
-                res = last_check_date + timedelta(days=frequency_in_days)
-            else:
-                last_date = datetime.strptime(
-                    last_check_date, "%Y-%m-%d").date()
-                res = last_date + timedelta(days=frequency_in_days)
+            res = last_check_date + timedelta(days=frequency_in_days)
             return res
 
+    @classmethod
+    def jobs_date_check(cls, jobs, last_check_date):
+        # Excludes old job posts by comparing their creation date with last check date
+        if isinstance(jobs, QuerySet):
+            for job in jobs:
+                if job.creation_date.date() <= last_check_date:
+                    jobs = jobs.exclude(id=job.id)
+            return jobs
+        return JobPost.objects.none()
+
+    @classmethod
+    def send_mail(cls, jobs, alert):
+        # Sends mail to the user to inform there are new relevant job posts after checking there are such job posts
+        if jobs.count() <= 0:
+            return
+        user = alert.user_account_id
+        user_mail = user.email
+        send_mail(
+            'job alert from joboard',
+            'Hi, there are new jobs that suits your requirements! you can check them out in the website',
+            settings.EMAIL_HOST_USER,
+            [user_mail],
+            fail_silently=False,
+        )
+
+    @classmethod
+    def check_new_jobs(cls):
```

Comment:
change the function name - you're not checking you're also doing something here...

Reply:
fixed @bardielle  @guy9050